src/app/qgisapp.cpp

```diff
@@ -10214,26 +10214,43 @@ void QgisApp::pasteFromClipboard( QgsMapLayer *destinationLayer )
     {
       newFeatures.clear();
 
-      QgsFixAttributeDialog *dialog = new QgsFixAttributeDialog( pasteVectorLayer, invalidFeatures, this );
-      int feedback = dialog->exec();
+      QgsAttributeEditorContext context( createAttributeEditorContext() );
+      context.setAllowCustomUi( false );
+      context.setFormMode( QgsAttributeEditorContext::StandaloneDialog );
 
-      switch ( feedback )
+      QgsFixAttributeDialog *dialog = new QgsFixAttributeDialog( pasteVectorLayer, invalidFeatures, this, context );
+
+      connect( dialog, &QgsFixAttributeDialog::finished, this, [ = ]( int feedback )
```

Comment:
This will leak if the dialog is cancelled

Reply:
Out of interest.

The [QDialog docs](https://doc.qt.io/qt-5/qdialog.html) mention

> e.g. an OK button to the accept() slot and a Cancel button to the reject() slot.
> [...]
> If the user presses the Esc key in a dialog, QDialog::reject() will be called.

and

> QDialog::finished(int result) : This signal is emitted when the dialog's result code has been set, either by the user or by calling done(), accept(), or reject().

... but I cannot exclude that there's a way to get out of the dialog without deleting it. What would be the proper solution?

Reply:
So does pressing cancel in this dialog still result in the paste happening?

Reply:
I assumed that's catched in the switch below, but @signedav is in the better position to answer.

Reply:
On canceling the dialog, `features` stays empty. Means we call `pasteFeatures` with 0 features and raise there the warning message that no features have  been copied. We could catch this case earlier, but it means to have a duplicate code for the message.

Reply:
Ok, fair enough!

Reply:
Thanks @nyalldawson @m-kuhn